ssi-ldp/src/lib.rs

```diff
@@ -469,6 +469,8 @@ impl LinkedDataProofs {
         key: &JWK,
         extra_proof_properties: Option<Map<String, Value>>,
     ) -> Result<Proof, Error> {
+        let mut options = options.clone();
+        ensure_or_pick_verification_relationship(&mut options, document, key, resolver).await?;
```

Comment:
Can you make this change in `prepare()` too?

Reply:
ah yeah, sorry changes vm, which is used in the `pick_proof_suite` call